src/vs/workbench/contrib/scm/browser/scmViewPane.ts

```diff
@@ -783,6 +798,8 @@ class ViewModel {
 
 		configurationService.onDidChangeConfiguration(this.onDidChangeConfiguration, this, this.disposables);
 		this.onDidChangeConfiguration();
+
+		this.disposables.add(this.tree.onDidChangeCollapseState(() => this.updateViewState()));
```

Comment:
This might be too expensive. We really only need to snapshot the view state whenever the view goes invisible or when the workbench shuts down.